reports-api/src/reports_api/resources/staff.py

```diff
@@ -94,19 +106,32 @@ class Staff(Resource):
     @profiletime
     def get(staff_id):
         """Return a staff detail based on id."""
-        return StaffService.find_by_id(staff_id), HTTPStatus.OK
+        staff = StaffService.find_by_id(staff_id)
+        if staff:
+            return staff, HTTPStatus.OK
+        return f"Staff with id '{staff_id}' not found", HTTPStatus.NOT_FOUND
 
     @staticmethod
-    @cors.crossdomain(origin='*')
+    @cors.crossdomain(origin="*")
     @auth.require
     @profiletime
     def put(staff_id):
         """Update and return a staff."""
-        staff = StaffService.update_staff(staff_id, API.payload)
-        return staff.as_dict(), HTTPStatus.OK
+        staff_schema = StaffSchema()
+        try:
+            request_json = staff_schema.load(API.payload)
+        except ValidationError as err:
+            return err.messages, HTTPStatus.BAD_REQUEST
+        exists = StaffService.check_existence(
```

Comment:
Existence check of staff should be done based on email address. There can be people with same Firstname and lastname

Reply:
Updated